src/vs/workbench/parts/files/electron-browser/fileActions.ts

```diff
@@ -1335,35 +1336,77 @@ export function validateFileName(parent: IFileStat, name: string, allowOverwriti
 		return nls.localize('emptyFileNameError', "A file or folder name must be provided.");
 	}
 
+	const names: string[] = trimTrailingSlashes(name)	// prevents empty last array element after split
+		.split(/[\\/]/);
+
+
 	// Do not allow to overwrite existing file
 	if (!allowOverwriting) {
-		if (parent.children && parent.children.some((c) => {
-			if (isLinux) {
-				return c.name === name;
+		let p = parent;
+
+		const alreadyExisting = names.every((folderName) => {
+			let { exists, child } = alreadyExists(p, folderName);
+
+			if (!exists) {
+				return false;
+			} else {
+				p = child;
+				return true;
 			}
+		});
 
-			return c.name.toLowerCase() === name.toLowerCase();
-		})) {
+		if (alreadyExisting) {
 			return nls.localize('fileNameExistsError', "A file or folder **{0}** already exists at this location. Please choose a different name.", name);
 		}
 	}
 
 	// Invalid File name
-	if (!paths.isValidBasename(name)) {
+	if (names.some((folderName) => !paths.isValidBasename(folderName))) {
 		return nls.localize('invalidFileNameError', "The name **{0}** is not valid as a file or folder name. Please choose a different name.", trimLongName(name));
 	}
 
-	// Max length restriction (on Windows)
 	if (isWindows) {
-		const fullPathLength = name.length + parent.resource.fsPath.length + 1 /* path segment */;
-		if (fullPathLength > 255) {
+		if (windowsMaxLengthRestrictionReached(name, parent)) {
 			return nls.localize('filePathTooLongError', "The name **{0}** results in a path that is too long. Please choose a shorter name.", trimLongName(name));
 		}
 	}
 
 	return null;
 }
 
+function windowsMaxLengthRestrictionReached(name: string, parent: IFileStat): boolean {
+	const fullPathLength = name.length + parent.resource.fsPath.length + 1 /* path segment */;
+	return fullPathLength > 255;
+}
+
+function trimTrailingSlashes(str): string | undefined {
+	if (!str) { return str; }
+	return str.replace(/[\\/]*$/, '');
+}
+
+function alreadyExists(parent: IFileStat, name: string): { exists: boolean, child: IFileStat | undefined } {
+	let foundDupChild: IFileStat;
+
+	if (parent.children) {
+		let exists: boolean = parent.children.some((c) => {
+			let found = compareFolderNames(c.name, name);
+			if (found) { foundDupChild = c; }
+			return found;
+		});
+		return { exists, child: foundDupChild };
+	}
+
+	return { exists: false, child: undefined };
+}
+
+function compareFolderNames(name1: string, name2: string): boolean {
```

Comment:
I do not really see the benefit of moving the folderName comparison to a seperate method. I would keep it as it iwas before, especially since compare functions usually return -1, 0 and 1 so for me this is confusing naming since it returns a boolean. If it returns a boolean it shoulld be areFolderNamesEqual. Though I prefer not having a method